index.js

```diff
@@ -0,0 +1,36 @@
+var btn = document.getElementById('action');
+
+btn.addEventListener('click', throttle(letsDoIt, 1000));
+
+function throttle(fn, time) {
+  var isThrottle = false;
+  return function () {
+
+    if (!isThrottle) {
+      fn();
+      isThrottle = true;
+      setTimeout(function () {
+        isThrottle = false;
+      }, time);
+    }
+
+    return;
```

Comment:
Mostly **throttle** returns the last result of the `fn` execution (a kind of memoization).